src/components/UploaderForm.tsx

```diff
@@ -0,0 +1,98 @@
+import { FunctionComponent, useState, useMemo } from 'react';
+import { Button, ErrorMessage } from 'lbh-frontend-react';
+import { Formik, Form } from 'formik';
+import * as Yup from 'yup';
+import UploaderPanel from './UploaderPanel';
+import { useRouter } from 'next/router';
+import { DocumentType } from '../domain/document-type';
+import { EvidenceRequest } from '../domain/evidence-request';
+
+const UploaderForm: FunctionComponent<Props> = (props) => {
+  const router = useRouter();
+  const { requestId } = router.query;
+  const [submitError, setSubmitError] = useState(false);
+
+  const requestedDocuments = [props.evidenceRequest.documentType];
+
+  const schema = useMemo(
+    () =>
+      Yup.object(
+        requestedDocuments.reduce(
+          (others, key) => ({
+            ...others,
+            [key.id]: Yup.mixed().required('Please select a file'),
+          }),
+          {}
+        )
+      ),
+    [requestedDocuments]
+  );
+
+  const initialValues = useMemo(
+    () =>
+      requestedDocuments.reduce(
+        (others, key) => ({
+          ...others,
+          [key.id]: '',
+        }),
+        {}
+      ),
+    [requestedDocuments]
+  );
+
+  console.log(initialValues);
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={schema}
+      onSubmit={async (values) => {
```

Comment:
This function will get recreated on every React render, try using the `useCallback` hook to create a memoised callback function we can use for this prop